### PR TITLE
dockerImages.gremlin: fluentd integration

### PR DESCRIPTION
**Pull request by eonpatapon**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -17,6 +17,10 @@ let pkgs = import nixpkgs {};
 
       perp = callPackage ./pkgs/perp { };
 
+      fluentd = callPackage ./pkgs/fluentd { };
```
Comment (nlewo) [inline default.nix]: Could you put this in a dedicated commit with a commit message:
```
fluentd: init at 0.14
```

File: pkgs/fluentdCwPlugins/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchgit }:
+
+stdenv.mkDerivation rec {
+  name = "fluentd-cloudwatt-plugins";
+
+  src = fetchgit {
+    rev = "0c418901bcf98b8e75f248ffe103f26eff65ef2f";
+    url = "https://git.corp.cloudwatt.com/docker/kubernetes-base.git";
```
Comment (nlewo) [inline pkgs/fluentdCwPlugins/default.nix]: Would it be possible to fetch them from a public location?

File: pkgs/docker-images/default.nix [hunk 3]
```diff
@@ -110,27 +110,52 @@ in
...
+      }
+      {
+        name = "fluentd";
         preStartScript = ''
-          # We can't modify the parent image, so we do it at runtime
-          if [ -f /etc/prometheus/prometheus_jmx_java8.yml ] && ! grep -q 'metrics<name'
-          then
-            echo "- pattern: 'metrics<name=(.+)><>(.+):'" >> /etc/prometheus/prometheus_jmx_java8.yml
-          fi
+          [ ! -p /tmp/gremlin-server ] && mkfifo /tmp/gremlin-server
```
Comment (nlewo) [inline pkgs/docker-images/default.nix]: Could you add a comment to say what are you doing :)
Since they are used in a configuration file, it can be hard to understand why you are creating them.

File: default.nix (hunk 1 above)
Comment (eonpatapon) [reply default.nix]: done

File: pkgs/fluentdCwPlugins/default.nix [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{ stdenv }:
+
+stdenv.mkDerivation rec {
```
Comment (nlewo) [inline pkgs/fluentdCwPlugins/default.nix]: I think we don't need this derivation:
If somewhere you use `fluentd -p ${fluentdCwPlugins/plugins/filter_generic_metadata.rb}`, the file is copyied to the store by Nix.
